cuda/test/base/cuda_executor.cu

```diff
@@ -176,6 +183,29 @@ TEST_F(CudaExecutor, CopiesDataToCuda)
 }
 
 
+__global__ void check_data2(int *data)
+{
+    if (data[0] != 4 || data[1] != 8) {
+        asm("trap;");
```

Comment:
Is it to make sure the compiler does not ignore this kernel?

Reply:
This is meant to fail if the data contains the incorrect value